src/extensions/unit/unitext_hooks.cpp

```diff
@@ -680,6 +681,285 @@ DECLARE_PATCH(_UnitClass_Per_Cell_Process_AutoHarvest_Assign_Harvest_Mission_Pat
+    newunit->FirepowerBias = this_ptr->FirepowerBias;
+    newunit->SpeedBias = this_ptr->SpeedBias;
+    newunit->Coord = this_ptr->Coord;
+    newunit->EMPFramesRemaining = this_ptr->EMPFramesRemaining;
+    newunit->Ammo = this_ptr->Ammo;
+
+
+    if (newunit->Unlimbo(newunit->Coord, this_ptr->PrimaryFacing.Current().Get_Dir())) {
+
+        /**
+         *  Unlimbo successful, select our new unit and return it
+         */
+
+        if (PlayerPtr == newunit->Owning_House()) {
+            newunit->Select();
+        }
+
+        if (this_ptr->TarCom) {
+            newunit->Assign_Target(this_ptr->TarCom);
+            newunit->Assign_Mission(MISSION_ATTACK);
+            newunit->Commence();
+        }
+
+        return newunit;
+    }
+
+    /**
+     *  Unlimboing the new unit failed! Delete the new unit and return false.
+     */
+    delete newunit;
+    return nullptr;
+}
+
+/**
+ *  Work-around function because the compiler likes smashing the stack by using ebp
+ *  when calling locomotor functions :)
+ *
+ *  Returns the address that the calling function should jump to after calling this.
+ *
+ *  @author: Rampastring
+ */
+int _UnitClass_Try_To_Deploy_Transform_To_Vehicle_Patch_Func(UnitClass* this_ptr) {
+
+    /**
+     *  Stolen bytes/code.
+     */
+    if (this_ptr->Class->DeploysInto != nullptr) {
+
+        /**
+         *  This unit is deployable rather than transformable, check whether it can deploy.
+         */
+        goto original_code;
+    }
+
+    UnitTypeClassExtension* unittypeext = Extension::Fetch<UnitTypeClassExtension>(this_ptr->Class);
+
+    if (unittypeext->TransformsInto != nullptr) {
+
+        /**
+         *  Use custom "transform to vehicle" logic if we don't need charge or we have enough of it.
+         */
+
+        if (unittypeext->IsTransformRequiresFullCharge && this_ptr->CurrentCharge < this_ptr->Class->MaxCharge) {
+
+            /**
+             *  We don't have enough charge, return false
+             */
+            return 0x006511A0;
+        }
+
+        this_ptr->Mark(MARK_UP);
+        this_ptr->Locomotor_Ptr()->Mark_All_Occupation_Bits(MARK_UP);
+
+        UnitClass* newunit = Create_Transform_Unit(this_ptr);
+
+        if (newunit != nullptr) {
+
+            /**
+             *  Creating transformed unit succeeded, erase the original unit and force function to return true
+             */
+            return 0x0065114C;
+        } else {
+
+            /**
+             *  Creating transformed unit failed. Re-mark our occupation bits and return false.
+             */
+            return 0x00651168;
+        }
+    }
+
+    /**
+     *  Continue to deployability check.
+     */
+original_code:
+    return 0x00650BC2;
+}
+
+
+/**
+ *  #issue-715
+ *
+ *  Transforms a unit to another unit when a transformable unit deploys.
+ *
+ *  @author: Rampastring
+ */
+DECLARE_PATCH(_UnitClass_Try_To_Deploy_Transform_To_Vehicle_Patch) {
+    GET_REGISTER_STATIC(UnitClass*, this_ptr, esi);
```

Comment:
Needs a space between the `UnitClass` and `*`.

This is most likely your MSVC auto-formatting taking over here, you can change those easily in Tools -> Options -> Text Editor -> C/C++.